extensions/interactions/ItemSelectionInput/protractor.js

```diff
@@ -0,0 +1,159 @@
+var objects = require('../../objects/protractor.js');
+var waitFor = require('../../../core/tests/protractor_utils/waitFor.js');
+
+// The members of richTextInstructionsArray are functions, one for each option,
+// which will each be passed a 'handler' that they can use to edit the
+// rich-text area of the option, for example by
+//   handler.appendUnderlineText('emphasised');
+var customizeInteraction = function(
+    elem, richTextInstructionsArray, maxSelectionAllowed) {
+  objects.IntEditor(
+    elem.all(by.tagName('schema-based-int-editor')).last()
+  ).setValue(maxSelectionAllowed);
+
+  forms.ListEditor(elem).setLength(richTextInstructionsArray.length);
+  for (var i = 0; i < richTextInstructionsArray.length; i++) {
+    var richTextEditor = forms.ListEditor(elem).editItem(i, 'RichText');
+    richTextEditor.clear();
+    richTextInstructionsArray[i](richTextEditor);
+  }
+};
+
+// These members of richTextInstructionsArray each describe how to check one of
+// the options.
+var expectInteractionDetailsToMatch = function(
+    elem, richTextInstructionsArray) {
+  elem.all(by.repeater('choice in $ctrl.choices track by $index'))
+    .then(function(optionElements) {
+      expect(optionElements.length).toEqual(richTextInstructionsArray.length);
+      for (var i = 0; i < optionElements.length; i++) {
+        forms.expectRichText(optionElements[i].element(by.css(
+          '.protractor-test-item-selection-option'
+        ))).toMatch(richTextInstructionsArray[i]);
+      }
+    });
+};
+
+// type of object returned by interaction
+var answerObjectType = 'SetOfHtmlString';
+
+// 'elem' is the HTML element containing the form to submit the answer to.
+// 'answer' Set([{String}]) is the text on the multiple-choice item to select.
+var submitAnswer = function(elem, answer) {
+  var answerArray = Array.from(answer);
+
+  for (var i = 0; i < answerArray.length; i++) {
+    var desiredAnswer = answerArray[i];
+    elem.element(by.cssContainingText(
+      '.protractor-test-item-selection-input-item', desiredAnswer))
+      .element(by.css(
+        '.protractor-test-item-selection-input-checkbox')).click();
+  }
+
+  var submitAnswerButton = element(by.css(
+    '.protractor-test-submit-answer-button'));
+  waitFor.elementToBeClickable(submitAnswerButton,
+    'Submit Answer button is not clickable');
+  submitAnswerButton.click();
+};
+
+var testSuite = [{
+  interactionArguments: [[function(editor) {
+    editor.appendBoldText('answer1');
+  }, function(editor) {
+    editor.appendItalicText('answer2');
+  }, function(editor) {
+    editor.appendItalicText('answer3');
+  }], 3],
+  ruleArguments: ['DoesNotContainAtLeastOneOf', ['answer1', 'answer2']],
+  expectedInteractionDetails: [[function(checker) {
+    checker.readBoldText('answer1');
+  }, function(checker) {
+    checker.readItalicText('answer2');
+  }, function(checker) {
+    checker.readItalicText('answer3');
+  }]],
+  wrongAnswers: [['answer1', 'answer2']],
```

Comment:
Is it possible to test submitting the wrong number of answers?

Reply:
By this, do you mean submitting more than the maximum possible number of answers? In that case, the expected behavior would be a warning stating that "No more than x answers may be selected." and remaining answer options becoming un-clickable. I'm not sure how we could integrate that check into the testing suite as it currently stands, but I could look into it?

Reply:
Okay, why don't you take a brief look and see if there's an obvious way to do it. If not, this isn't critical

Reply:
I don't think there's an obvious way to do it since the exports of the interaction don't include the maximum number of selections allowed and there's no easy way to access it (so we couldn't just do an "if answerArrayLength > maxSelectionsAllowed, expect warning on screen & checkbox not to be clickable", and I think without such an if statement we'd just end up with an error that the checkboxes aren't clickable).

Reply:
The worst case scenario if this functionality breaks because we didn't test it is:
1. The user submits too many answers and is marked wrong. This should be fine, as the answer is indeed wrong.
2. The question doesn't specify the expected number of selected options, and more than the set max are valid. The user might get quite confused here trying to figure out how many options are expected, but this would be a very badly worded question.